hybrasyl/Objects/Creature.cs

```diff
@@ -109,12 +221,272 @@ public Creature()
+                        case IntentDirection.Right:
+                            {
+                                switch (Direction)
+                                {
+                                    case Direction.North:
+                                        {
+                                            //facing north, attack east
+                                            rect = new Rectangle(X, Y, 1 + intent.Radius, 1);
+                                        }
+                                        break;
+                                    case Direction.South:
+                                        {
+                                            //facing south, attack west
+                                            rect = new Rectangle(X - intent.Radius, Y, intent.Radius, 1);
+                                        }
+                                        break;
+                                    case Direction.East:
+                                        {
+                                            //facing east, attack south
+                                            rect = new Rectangle(X, Y - intent.Radius, 1, intent.Radius);
+                                        }
+                                        break;
+                                    case Direction.West:
+                                        {
+                                            //facing west, attack north
+                                            rect = new Rectangle(X, Y, 1, 1 + intent.Radius);
+                                        }
+                                        break;
+                                }
+                            }
+                            break;
+                        case IntentDirection.Nearby:
+                        case IntentDirection.None:
+                            {
+                                //attack radius
+                                rect = new Rectangle(X - intent.Radius, Y - intent.Radius, Math.Max(intent.Radius, (byte)1), Math.Max(intent.Radius, (byte)1));
+                            }
+                            break;
+                    }
+
+                    if (!rect.IsEmpty)
+                    {
+                        var intersect = Map.EntityTree.GetObjects(Rect).Where(obj => obj is Creature && obj != this);
+                        // Disqualify targets based on intent
+                        // If we aren't targeting friendlies or pvp, remove all users entirely
+                        if (!intent.Target.Contains(IntentTarget.Friendly) && !intent.Target.Contains(IntentTarget.Pvp))
+                            intersect = intersect.Where(e => !(e is User));
+
+                        // If we aren't targeting hostiles, remove all monsters
+                        if (!intent.Target.Contains(IntentTarget.Hostile))
+                            intersect.Where(e => !(e is Monster || (e as Creature).Condition.PvpEnabled));
```

Comment:
After this line, I don't see anything that adds the intersect targets to the possibleTargets List.

Reply:
This has been refactored (and verified to work).